syntax = "proto3";

enum StringFormatterStatus {
	FAILURE = 0;
	SUCCESS = 1;
}

// StringFormatter service definition. 
service StringFormatterService {
	// Formats the string.	
	rpc formatInt(IntStringFormatterInput) returns (StringFormatterOutput) { }
}

message IntStringFormatterInput {
	string string_to_format = 1;
	int32 num0 = 2;
	int32 num1 = 3;
	int32 num2 = 4;
}

message StringFormatterOutput {
	StringFormatterStatus status = 1;
	string result = 2;
}
